blocksync/buffer_test.go

```diff
@@ -132,7 +137,12 @@ func TestBlockBufferGetBlocksIntervalsToSync(t *testing.T) {
 	registry := protocol.NewRegistry()
 	rp := rolldpos.NewProtocol(cfg.Genesis.NumCandidateDelegates, cfg.Genesis.NumDelegates, cfg.Genesis.NumSubEpochs)
 	require.NoError(rp.Register(registry))
-	sf, err := factory.NewFactory(cfg, factory.InMemTrieOption(), factory.RegistryOption(registry))
+	factoryCfg := factory.Config{
+		DB:      cfg.DB,
+		Chain:   cfg.Chain,
+		Genesis: cfg.Genesis,
```

Comment:
add a new fun for this?
```
func GenerateConfig(cfg *config.Config) factory.Config
```
so can simplify tons of this in the PR

Reply:
> add a new func for this?
> 
> ```
> func GenerateConfig(cfg *config.Config) factory.Config
> ```
> 
> so can simplify tons of this in the PR

will cause `import cycle` errors

Reply:
oh ok